### PR TITLE
feat!: intmdb 1023 private endpoint service

### PR DESCRIPTION
**Pull request by Adelmar92**
## Proposed changes

This pull request updates the next CDKs
- Private Endpoint AWS L1 - new resource
- Private Endpoint Service L1 - new resource
- Atlas Basic L3 private endpoint - modified to represent the CFN resources
_Jira ticket:_ [INTMDB-1023](https://jira.mongodb.org/browse/INTMDB-1023)

CFN pull request : [704](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/704)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code
- [ ] I have tested the CDK constructor in a CFN stack. See [TESTING.md](../TESTING.md)

### CODE REVIEW

File: src/l1-resources/private-endpoint/index.ts [hunk 1]
```diff
@@ -15,134 +15,75 @@ export interface CfnPrivateEndpointProps {
...
 
   /**
-   * List of string representing the AWS VPC Subnet ID (like: subnet-xxxxxxxxxxxxxxxxx) (Used For Creating the AWS VPC Endpoint)
+   * Cloud service provider that manages this private endpoint.
    *
-   * @schema PrivateEndpoint#SubnetIds
+   * @schema CfnPrivateEndpointProps#CloudProvider
    */
-  readonly subnetIds?: string[];
+  readonly cloudProvider?: string;
```
Comment (andreaangiolillo) [inline src/l1-resources/private-endpoint/index.ts:44]: FYI: this is a breaking change which means that we need to increase the Major version of the cdk. I updated the title and description of the ticket to make this clear

File: examples/l1-resources/private-endpoint.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
+import { CfnPrivateEndpointAws } from './../../src/l1-resources/private-endpoint-aws/index';
 import * as cdk from 'aws-cdk-lib';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
 import { Construct } from 'constructs';
-import { CfnPrivateEndpoint } from 'awscdk-resources-mongodbatlas';
+import { CfnPrivateEndpointService, CfnPrivateEndpointServicePropsCloudProvider } from '../../src/l1-resources/private-endpoint-service';
```
Comment (AgustinBettati) [inline examples/l1-resources/private-endpoint.ts:5]: given that the old private-endpoint resource was not removed, should we leave this example as it was originally and add 2 new examples for the new resources (private-endpoint-service and private-endpoint-aws)?

File: examples/l3-resources/atlas-basic-private-endpoint.ts [hunk 3]
```diff
@@ -53,13 +53,8 @@ export class CdkTestingStack extends cdk.Stack {
...
       privateEndpointProps: {
-        privateEndpoints: [
-          {
-            vpcId: atlasProps.vpcId,
-            subnetIds: [atlasProps.subnetId],
-          }
-  
-        ]
+        awsVpcId: atlasProps.vpcId,
+        awsSubnetId: atlasProps.subnetId,
```
Comment (AgustinBettati) [inline examples/l3-resources/atlas-basic-private-endpoint.ts:57]: is this a breaking change in the l3 constructor?

File: src/l3-resources/atlas-basic-private-endpoint/index.ts [hunk 4]
```diff
@@ -84,7 +118,7 @@ export interface AtlasBasicPrivateEndpointProps {
    * @default us-east-1
    * @memberof AtlasPrivateEndpointProps
    */
-  readonly region?: string;
+  readonly region: string;
```
Comment (AgustinBettati) [inline src/l3-resources/atlas-basic-private-endpoint/index.ts:121]: why is this prop now required?

File: examples/l1-resources/private-endpoint.ts (hunk 2 above)
Comment (Adelmar92) [reply examples/l1-resources/private-endpoint.ts:5]: i dont like the idea of leaving the original, ill just rename it to V1-deprecated

File: src/l3-resources/atlas-basic-private-endpoint/index.ts (hunk 4 above)
Comment (Adelmar92) [reply src/l3-resources/atlas-basic-private-endpoint/index.ts:121]: it should be a mandatory field, it was throwing me an error when i did not add it

**Review by AgustinBettati — APPROVED** [review]
LGTM, make sure to publish with a major version due to breaking changes in the l3 constructor

**Pull request merged**